app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java

```diff
@@ -1,3 +1,156 @@
-package ar.com.wolox.android.training;
+package ar.com.wolox.android.training.ui.login;
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
 
-public class LogInPresenterTest {}
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import ar.com.wolox.android.training.ui.network.entities.UserResponse;
+import ar.com.wolox.android.training.ui.network.services.LogInService;
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+
+public class LogInPresenterTest {
+
+    RetrofitServices retrofitServicesMock;
+    LogInService logInServiceMock;
+    LogInView mLoginViewMock;
+    LogInPresenter logInPresenter;
+
+    @Before
+    public void beforeTest(){
+        mLoginViewMock = Mockito.mock(LogInView.class);
+        retrofitServicesMock = Mockito.mock(RetrofitServices.class);
+        logInServiceMock = Mockito.mock(LogInService.class);
+
+        Mockito.when(retrofitServicesMock.getService(eq(LogInService.class))).thenReturn(logInServiceMock);
+        logInPresenter = new LogInPresenter(mLoginViewMock, retrofitServicesMock);
+    }
+
+    @Test
+    public void reposForUserOnFailure(){
+        // Given
+        Call<List<UserResponse>> callMock = Mockito.mock(Call.class);
+        Mockito.when(logInServiceMock.reposForUser(anyString())).thenReturn(callMock);
+        Mockito.doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Callback callback = invocation.getArgument(0);
+                callback.onFailure(callMock, new Exception("Hola"));
+                //callback.onResponse(null, null);
+                return null;
+            }
+        }).when(callMock).enqueue(any(Callback.class));
+
+        // When
+        logInPresenter.doLogin("asdasdasdasd", "afrafea");
```

Comment:
I don't care that much about theses texts because we are not using it. But just in case we should use easy to read strings and not something "asdasddasd" because the client could be reviewing the code. Just commenting.